### DISCUSSION TITLE
Is there a way to insert a page break into a code listing?

### DISCUSSION THREAD

**Discussion by BruceEckel** [marp-team/marp#346] [Q&A]
So code listings can be easily broken up into individual slides.

Ideally this would be in the form of a single-line comment that doesn't appear in the generated slide, so the line-break directives can be embedded into the original source-code files. (This allows the code to be updated and tested and then refreshed into the slides).

For example:

```
// <!-- pageBreak -->
```
or for a language with a different comment syntax like Python:
```
# <!-- pageBreak -->
```

Sorry if this exists and I missed it; I searched around.

**Comment by yhatt** [comment]
What is meaning of "code listing"? Markdown's [list syntax](https://www.markdownguide.org/basic-syntax/#lists-1), or [code blocks](https://www.markdownguide.org/basic-syntax/#code-blocks)?

In either case, I think you have to [write a Marp(it) plugin](https://marpit-api.marp.app/) for getting your idea. In Marp and Marpit, `---` ruler and [`#` headings with `headingDivider` global directive](https://marpit.marp.app/directives?id=heading-divider) are only way to split pages.

**Comment by BruceEckel** [comment]
Sorry I wasn't clear. I mean a computer programming source-code listing, for example:

```C++
#include <string>
#include <iostream>
using namespace std;

void f(string s) {
    cout << "f(string s): " << s << endl;
}
// {pageBreak} -- or whatever tag does the job
void f(float f) {
    cout << "f(float f): " << f << endl;
}

int main() {
    f("x");
    f(1.0);
}
```

Thanks for the link to the plugin docs. While that looks fascinating, I'm not in a position right now to delve in and figure it out. Is there any kind of "bounty" I could pay to get you to write the plugin for me? Thanks!

**Reply by yhatt** [reply]
Sorry, I don't want to be paid for a custom-made plugin for specific feature [by some reasons](https://opensource.guide/getting-paid/#why-some-people-seek-financial-support:~:text=There%20are%20many%20reasons%20why%20a%20person%20would%20not%20want%20to%20be%20paid%20for%20their%20open%20source%20work.).

**Reply by BruceEckel** [reply]
Didn't mean to offend. Still vote for the feature however it happens.

**Comment by BruceEckel** [comment]
PS: I understand if you cannot do this. I do think it would be a great addition to Marp for people who give programming presentations!
